pkg/prometheus/watcher.go

```diff
@@ -255,6 +280,11 @@ func (w *Watcher) addHandlers() {
 		AddFunc:    w.handleAlertmanagerAdd,
 		DeleteFunc: w.handleAlertmanagerDelete,
 	})
+	w.amcInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    w.handleAlertmanagerConfigAdd,
+		UpdateFunc: w.handleAlertmanagerConfigUpdate,
```

Comment:
Updatefunc is required since the alert manager config can be modified, the secret needs to be fixed accordingly.